pagopa/io-app
# fix: [IA-850] In develop mode, during the payment flow, if you go back, an error alert is displayed

## Short description
This pr removes an error alert displayed in develop mode if you go back during the payment flow.

Before:

https://user-images.githubusercontent.com/26501317/166218271-83b283f1-5eaf-4b8a-b276-dd58631f2d9e.mov


After:

https://user-images.githubusercontent.com/26501317/166217766-3b2ec4ee-4fa5-4612-a370-c27eab0122e7.mov


## List of changes proposed in this pull request
- Removed obsolete logic (sadly just a part) from `watchBackToEntrypointPaymentSaga`

## How to test
- Have in wallet bancomatPay only
- Messages 
- Tap on payment message
- Tap on "Display notice"
- Tap on "Go to payment"
- Tap on "Back"
- Tap on "Cancel"
- Now the navigation error is not displayed  👀
